desktop/cmp/input/TextInput.js

```diff
@@ -117,32 +117,34 @@ class Model extends HoistInputModel {
 
 const cmp = hoistCmp.factory(
     ({model, className, ...props}, ref) => {
-        const {width, flex, ...layoutProps} = getLayoutProps(props);
+        const {width, flex, ...layoutProps} = getLayoutProps(props),
+            // remove layoutProps, props not recognized as DOM elements, and props not accepted on BP's inputGroup component
+            {bind, commitOnChange, enableClear, onCommit, selectOnFocus, textAlign, ...rest} = getNonLayoutProps(props);
 
         const isClearable = !isEmpty(model.internalValue);
 
         return div({
             item: inputGroup({
+                ...rest,
+
+                // overwrite some of the ...rest props with Hoist's customizations
                 value: model.renderValue || '',
 
-                autoComplete: withDefault(props.autoComplete, props.type === 'password' ? 'new-password' : 'nope'),
-                autoFocus: props.autoFocus,
-                disabled: props.disabled,
-                inputRef: composeRefs(model.inputRef, props.inputRef),
-                leftIcon: props.leftIcon,
-                placeholder: props.placeholder,
-                rightElement: props.rightElement ||
-                    (props.enableClear && !props.disabled && isClearable ? clearButton() : null),
-                round: withDefault(props.round, false),
-                spellCheck: withDefault(props.spellCheck, false),
-                tabIndex: props.tabIndex,
-                type: props.type,
-
-                id: props.id,
+                autoComplete: withDefault(rest.autoComplete, rest.type === 'password' ? 'new-password' : 'nope'),
+                inputRef: composeRefs(model.inputRef, rest.inputRef),
+                rightElement: rest.rightElement ||
+                    (enableClear && !rest.disabled && isClearable ? clearButton() : null),
+                round: withDefault(rest.round, false),
+                spellCheck: withDefault(rest.spellCheck, false),
+
+                // These styles get passed onto the HTML INPUT element, 
+                // they are not placed on the DIV created by Blueprint's InputGroup component.
                 style: {
-                    ...props.style,
+                    ...rest.style,
                     ...layoutProps,
-                    textAlign: withDefault(props.textAlign, 'left')
+                    width: withDefault(width, 200),
+                    flex: withDefault(flex, null),
+                    textAlign: withDefault(textAlign, 'left')
```

Comment:
I am confused by some of these being passed here in the `inputGroup.style` block and then also in the style block below for the outer wrapper div we are installing.  (We have spent a lot of time on these inputs and layout issues in the past, so I'm on the lookout for any possible issues or regressions.)

- Does it make sense to specify flex on the input? Is it in a flex parent container?
- OK to have the width of the input and the width of the outer wrapper be exactly the same? Could the wrapper have padding or borders or anything else?
  - Note also that there is an `InputGroup.fill` prop - if we are deciding the wrap in a div and manage the width there, perhaps this should always be set to true for the inner input group?

These longstanding sizing/layout questions have always been a pain point for me, and I hate to have to bring them into this PR.  If we can use this to clarify them and get a better handle on the layouts, that would be excellent. If we are introducing more uncertainty, however, perhaps there is a way to restructure this change to get the other props passed through that we want without needing to rock the boat re. layout....